### PR TITLE
feat(rulesets): initial rulesets for the Arazzo Specification

### PR DESCRIPTION
**Pull request by frankkilcommins**
Fixes #[ISSUE_NUMBER].

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

New ruleset for the Arazzo Specification as well as appropriate docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Spectral, being a generic YAML/JSON linter, **needs a ruleset** to lint files. A
 To get started, run this command in your terminal to create a `.spectral.yaml` file that uses the Spectral predefined rulesets based on OpenAPI or AsyncAPI:
```
Comment (mnaumanali94) [inline README.md]:
```suggestion
To get started, run this command in your terminal to create a `.spectral.yaml` file that uses the Spectral predefined rulesets based on OpenAPI, Arazzo or AsyncAPI:
```

File: packages/rulesets/src/arazzo/functions/arazzoWorkflowIdUniqueness.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export default function arazzoWorkflowIdUniqueness(targetVal: ArazzoSpecification, _options: null): IFunctionResult[] {
+  const results: IFunctionResult[] = [];
+  const workflows = getAllWorkflows(targetVal);
+
+  const seenIds: Set<string> = new Set();
+  for (const { path, workflow } of workflows) {
+    const workflowId = workflow.workflowId;
+    if (seenIds.has(workflowId)) {
+      results.push({
+        message: `"workflowId" must be unique across all workflows.`,
```
Comment (mnaumanali94) [inline packages/rulesets/src/arazzo/functions/arazzoWorkflowIdUniqueness.ts]: Can we return a message saying which workflow isn't unique?

File: packages/rulesets/src/arazzo/index.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+      message: '{{error}}',
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoDocumentSchema,
+      },
+    },
+    'arazzo-workflowId-unique': {
+      description: 'Every workflow must have unique "workflowId".',
```
Comment (mnaumanali94) [inline packages/rulesets/src/arazzo/index.ts:31]: Based on above comment, return error here?

File: packages/rulesets/src/arazzo/index.ts [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+      recommended: true,
+      severity: 'warn',
+      given: '$',
+      then: {
+        field: 'info.description',
+        function: truthy,
+      },
+    },
+    'arazzo-info-summary': {
+      description: 'Info "summary" should be present and non-empty string.',
```
Comment (mnaumanali94) [inline packages/rulesets/src/arazzo/index.ts]: Description says should be present but it is a hint? Maybe change it to something like its recommended to have a summary?

File: packages/rulesets/src/arazzo/index.ts [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+      recommended: true,
+      severity: 0,
+      given: '$',
+      then: {
+        function: arazzoStepFailureActionsValidation,
+      },
+    },
+    'arazzo-step-success-actions-validation': {
+      description:
+        'Every success action must have a unique "name", and the fields "workflowId" and "stepId" are mutually exclusive.',
```
Comment (mnaumanali94) [inline packages/rulesets/src/arazzo/index.ts:87]: These could be two different rules IMO but fine like this too. The second part could be done with XOR function.

File: packages/rulesets/src/arazzo/index.ts [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+      given: '$.workflows[*].steps[*]',
+      then: {
+        field: 'description',
+        function: truthy,
+      },
+    },
+    'arazzo-step-summary': {
+      description: 'Step "summary" should be present and non-empty string.',
+      recommended: true,
+      severity: 'hint',
```
Comment (mnaumanali94) [inline packages/rulesets/src/arazzo/index.ts:207]: Description says should be present but it is a hint? Maybe change it to something like its recommended to have a summary?

File: packages/rulesets/src/arazzo/index.ts [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+        field: 'stepId',
+        function: pattern,
+        functionOptions: {
+          match: '^[A-Za-z0-9_\\-]+$',
+        },
+      },
+    },
+    'arazzo-step-operationPath': {
+      description: 'It is recommended to use "operationId" rather than "operationPath".',
+      recommended: true,
```
Comment (mnaumanali94) [inline packages/rulesets/src/arazzo/index.ts]: Remove all recommended  as it is true anyway

File: packages/rulesets/src/arazzo/functions/arazzoWorkflowIdUniqueness.ts (hunk 2 above)
Comment (frankkilcommins) [reply packages/rulesets/src/arazzo/functions/arazzoWorkflowIdUniqueness.ts]: Sure

File: packages/rulesets/src/arazzo/index.ts (hunk 5 above)
Comment (frankkilcommins) [reply packages/rulesets/src/arazzo/index.ts:87]: It's all wrapped up in the custom function for now at least. Some of it is captured by the JSON Schema check too

File: packages/rulesets/src/arazzo/index.ts (hunk 3 above)
Comment (frankkilcommins) [reply packages/rulesets/src/arazzo/index.ts:31]: added

**Review by mnaumanali94 — DISMISSED** [review]
🚀🚀🚀

**Review by mnaumanali94 — APPROVED** [review]
🎉

**Pull request merged**